### ISSUE TITLE
Lin. correction values are lost after reset

### ISSUE THREAD

**Issue by knoopx** [prusa3d/Prusa-Firmware#1371]
They are not properly saved to eeprom (broken since 3.4.1)

**Comment by Pathogenius** [comment]
My experience is with ECOR correction feature only. I recall printing the calibration object, and seeing clearly a favourable value for entering into MK3 menu. With a later firmware version, however, when I tried printing the same object, there was no clear value to choose.

A possibly anecdotal observation is that, the default value of 0 now produces equal or superior print results as compared to previous firmware (the one where I needed to use an ECOR value to get best results). My assumption is that there is no actual need for ECOR value, and that somehow the firmware has solved the issue without the need for calibration.

**Comment by zoltanf** [comment]
Same here, printed the ECOR tower and lowest (off) was best. I presume they will remove this menu item and the whole concept of linearity correction from future firmware versions.

**Comment by hoppke** [comment]
Might be valuable to people who run prusa machines with non-stock setups though. Probably no need for it in the lcd menu, but as a firmware feature it's still nice to have (when it works).

**Comment by aemileski** [comment]
Found the problem.  The code checks menu_leaving but not LCD_CLICKED. 

Bad code:
```
ultralcd.cpp 4734: if(menu_leaving)
```

Good example:
```
ultralcd.cpp 2875:  if (LCD_CLICKED || menu_leaving)
```

Interim fix:  If you let the menu timeout, it will save.

Also related, old non-existent function (was renamed lcd_settings_linearity_correction_menu_save but that already has a prototype at line 129):
```
ultralcd.cpp 121:  #ifdef LINEARITY_CORRECTION
ultralcd.cpp 122:  static void lcd_settings_menu_back();    
ultralcd.cpp 123:  #endif //LINEARITY_CORRECTION
```

I suspect the code is sufficiently broken that bad things will happen without linearity correction (at least 3 different related enable macros are used, and E-only probably no longer supported, etc.).

**Comment by aemileski** [comment]
Still a bug in 3.7.0 released 2019-04-01

Seriously?  I posted fix 2019-01-13

Here it is again, at new offset: 

Broken code:
```
ultralcd.cpp 5170:	if(menu_leaving)
ultralcd.cpp 5171:	{
ultralcd.cpp 5172:	    lcd_settings_linearity_correction_menu_save();
ultralcd.cpp 5173:	}
```

Fixed Code:
```
ultralcd.cpp 5170:	if(LCD_CLICKED || menu_leaving)
ultralcd.cpp 5171:	{
ultralcd.cpp 5172:	    lcd_settings_linearity_correction_menu_save();
ultralcd.cpp 5173:	}
```

**Comment by GurliGebis** [comment]
@aemileski may I suggest you create a pull request for it?
Since it is the official way to submit changes, it might get them to merge it (having it where it should be, instead of them having to dig through issues for code changes) :)

**Comment by 3d-gussner** [comment]
@knoopx Is this still an issue? I think it has been fixed in a newer firmware version, please confirm and close this issue.
[stale][fixed]

**Comment by knoopx** [comment]
@3d-gussner I don't know, I moved to klipper years ago because of the instability and lack of open development and never looked back again.

**Comment by 3d-gussner** [comment]
Good luck with Klipper sounds very interesting.🤞🍀
Can you please close this issue? Always nicer if the owner close issues instead of a forced close.😉

**Comment by aemileski** [comment]
@knoopx @3d-gussner This was finally fixed in v3.7.1 (May 13th, 2019) 

Note that open issue #1585 is a duplicate (I also reported that Mar 1st, 2019).

Yes, please close them both.
